### PR TITLE
Report Generation

### PR DESCRIPTION
**Pull request by saladenicoise**
Button on mainPage leads to report generation. Will only work if user is logged in as Admin

Two Forms
------------
First: Select a course from the list of courses currently found in the db and generates a report file

Second: Input user id and retrieve all courses registered and their data and generates a report file

Report file is automatically downloaded to the machine

### CODE REVIEW

File: data/reportGeneration/studentToCoursesScript.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    require_once('../../business/login/passwordUtils.php');
+
+    $conn = new mysqli(DB_SERVER, DB_USER, DB_PASS, DB_NAME);
+
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
+
+    $date = date('d-m-y-s');
+    $file = "report_courseList". $date .".txt";
```
Comment (smarkandu) [inline data/reportGeneration/studentToCoursesScript.php:13]: For the course report, I'd recommend including the course code as part of the filename

File: data/reportGeneration/courseToStudentsScript.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    // Downloads Report File
+    header('Content-Description: File Transfer');
+    header('Content-Disposition: attachment; filename='.basename($file));
+    header('Expires: 0');
+    header('Cache-Control: must-revalidate');
+    header('Pragma: public');
+    header('Content-Length: ' . filesize($file));
+    header("Content-Type: text/plain");
+    readfile($file);
+    unlink("report_studentList". $date .".txt");
```
Comment (smarkandu) [inline data/reportGeneration/courseToStudentsScript.php:45]: For the student report, I'd recommend including the student id as part of the filename

File: data/reportGeneration/studentToCoursesScript.php (hunk 1 above)
Comment (saladenicoise) [reply data/reportGeneration/studentToCoursesScript.php:13]: Good idea, will do.

**Review by smarkandu — COMMENTED** [review]
For both reports if we'e going the "text" approach, I'd recommend formatting it as a .CSV file (including file extension) such that it can be opened (are parsed) with Excel, etc.

[Nice to Have] For the report filenames, maybe include the timestamp as well.

**Review by smarkandu — APPROVED** [review]

**Pull request merged**